backend/src/utils/proxy.ts

```diff
@@ -70,53 +71,71 @@ export const proxyService =
         const namespace = service.namespace ?? getParam(request, 'namespace');
         const name = getParam(request, 'name');
 
-        const doServiceRequest = () => {
+        const getServiceAddress = async () => {
           const scheme = tls ? 'https' : 'http';
+          const serviceName = `${service.prefix ?? ''}${name}${service.suffix ?? ''}`;
+          if (DEV_MODE) {
+            // Use port forwarding for local development:
+            // kubectl port-forward -n <namespace> svc/<service-name> <local.port>:<service.port>
+            return `${scheme}://${local.host}:${local.port}`;
+          }
+          const annotation = SERVICE_ADDRESS_ANNOTATIONS[service.addressType];
+          if (annotation) {
+            try {
+              const k8sService = (
+                await fastify.kube.coreV1Api.readNamespacedService(serviceName, namespace)
+              ).body;
+              const address = k8sService.metadata.annotations[annotation];
+              if (address) {
+                return address;
+              }
+              fastify.log.error(
+                `Proxy could not find address annotation on k8s service ${serviceName} in namespace ${namespace}, falling back to internal address. Annotation expected: ${annotation}`,
+              );
+            } catch (e) {
+              fastify.log.error(
+                e,
+                `Proxy failed to read k8s service ${serviceName} in namespace ${namespace}, falling back to internal address`,
+              );
```

Comment:
Shouldn't this just fail instead of fall back; what's the use case?

Reply:
Hm, fair point. I don't think there is one, I'll change this to fail.

Reply:
Done